pkg/securelaunch/helpers.go

```diff
@@ -0,0 +1,270 @@
+ * 1. target file path where file was written
+ * 2. error
+ */
+func WriteToFile(data []byte, dst, defFileName string) (string, error) {
+
+	// make sure dst is an absolute file path
+	if !filepath.IsAbs(dst) {
+		return "", fmt.Errorf("Error: dst =%s Not an absolute path ", dst)
+	}
+
+	// target is the full absolute path where []byte will be written to
+	target := dst
+
+	/* dst = /foo/bar/eventlog, check if /foo/bar/eventlog is a dir or file that exists..
+	     * if it exists as a file, this if loop is untouched.
+		 * if it exists as a dir, append defFileName to it
+	     * NOTE both dst = /foo/bar/ and dst=/foo/bar are considered dirs i.e. trailing "/" has no meaning..
+	*/
+	dstFound, is_dir, err := exists(dst)
+	if is_dir {
+		if !dstFound {
+			return "", fmt.Errorf("destination dir doesn't")
+		} else {
+			// User provided a dir, Use defFileName as target
+			Debug("No file name provided. Adding it now. old target=%s", target)
+			target = filepath.Join(dst, defFileName)
+			Debug("New target=%s", target)
+		}
+	}
+
+	Debug("target=%s", target)
+	err = ioutil.WriteFile(target, data, 0644)
+	if err != nil {
+		return "", fmt.Errorf("Could't write file to %s, err=%v", target, err)
+	}
+	Debug("WriteToFile exit w success data written to target=%s", target)
+	return target, nil
+}
+
+/*
+ * src and dst are already mounted file paths.
+ * src and dst are Absolute file paths
+ * defFileName is default dst file name, only used if user doesn't provide one.
+ * returns
+ * 1. target file path where file was written
+ * 2. error
+ */
+func WriteSrcFiletoDst(src, dst, defFileName string) (string, error) {
+	srcFileFound, _, err := exists(src)
+	if !srcFileFound {
+		return "", fmt.Errorf("File to be written doesnt exist:")
+	}
+
+	d, err := ioutil.ReadFile(src)
+	if err != nil {
+		return "", fmt.Errorf("Error reading src file %s, err=%v", src, err)
+	}
+
+	target, err := WriteToFile(d, dst, defFileName)
+	if err != nil {
+		return "", fmt.Errorf("WriteSrcFiletoDst failed err=%v", err)
+	}
+
+	Debug("WriteSrcFileToDst exit w success src=%s written to target=%s", src, target)
+	return target, nil
+}
+
+/*
+ * NOTE: Caller's responsbility to unmount this..use return var mountPath to unmount in caller.
+ *
+ * obtains absolute file path based on input entered by user in policy file.
+ * inputVal is of format <block device identifier>:<path>
+ * inputVal e.g sda:/boot/foo.go , 4qccd342-12zr-4e99-9ze7-1234cb1234c4:/bar/zyx.go
+ * rw_option = true --> mount is of type read-write, else read-only.
+ *
+ * This function
+ * parses user input.
+ * mounts device
+ * Get absolute file path on mounted device.
+ * does NOT Unmount device, use returned mountPath to unmount...
+ * returns filePath, mountPath, error in that order
+ */
+func GetMountedFilePath(inputVal string, rw_option bool) (string, string, error) {
+	s := strings.Split(inputVal, ":")
+	if len(s) != 2 {
+		return "", "", fmt.Errorf("%s: Usage: <block device identifier>:<path>", inputVal)
```

Comment:
this error message doesn't seem to have anything to do with this library... this looks like cmdline args?

Reply:
All the callers of GetMountedFilePath use 
sda:/path/to/file _OR_ UUID:/path/to/file 
in the JSON policy file as inputs... Here is small snippet of the policy file:

```
{
    "default_action": "",
    "collectors": [
      {
        "type": "dmi",
        "events": [
          {
            "label": "BIOS",
            "fields": []
          },
        ]
      },
       {
            "type": "cpuid",
            "location": "sda1:/simran/"
       }
    ],
    "attestor": {},
    "launcher": {
        "type": "kexec",
        "params": {
            "initrd":"sdc1:/boot/initramfs-4.14.35-1941.el7uek.x86_64.img",
            "cmdline":"BOOT_IMAGE=/boot/vmlinuz-4.14.35-1941.el7uek.x86_64 root=/dev/mapper/ol-root ro crashkernel=auto rd.lvm.lv=ol/root netroot=iscsi:@10.196.210.62::3260::iqn.1986-03.com.sun:ovs112-boot rd.iscsi.initiator=iqn.1988-12.com.oracle:ovs112 rd.lvm.lv=ol/home rd.lvm.lv=ol/swap ip=eno2:dhcp console=ttyS0,115200n8 earlyprintk=serial,keep printk.time=y intel_iommu=on pci=realloc iommu=pt LANG=en_US.UTF-8",
            "kernel":"sdc1:/boot/vmlinuz-4.14.35-1941.el7uek.x86_64"
         }
    },
    "eventlog": {
        "type": "file",
        "location": "sda1:/evtlog"
    }
}
``` 

All the callers have inputs of the type where device identifier is separated from the file location by a :
Hence it made sense to parse the input in this function only since all sub packages were calling this...